src/main/java/ch/naviqore/service/Leg.java

```diff
@@ -0,0 +1,22 @@
+package ch.naviqore.service;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface Leg {
+    @NotNull Location getDepartureLocation();
+
+    @NotNull Location getArrivalLocation();
+
+    @Nullable Stop getDepartureStop();
+
+    @Nullable Stop getArrivalStop();
+
+    @NotNull DepartureTime getDepartureTime();
```

Comment:
Just use normal `LocalTime` here?

Reply:
This was mainly an idea to allow method overrides for getConnections() / earliestArrivals() specifying the direction of travel. But this could of course also be moved to the QueryConfig interface.